docs/onboarding.md

```diff
@@ -0,0 +1,157 @@
+[Table of contents](./README.md#table-of-contents)
+
+Onboarding
+==========
+
+This document explains the architecture and process allowing a cozy instance owner to register to its cozy instance.
+
+Compatibility with the current developments on cozy onboarding is a goal : [The following documents has been consulted for this proposal]( https://github.com/cozy/cozy-proxy/blob/bf9af7f2342e3fc183a8b4e72bcedb909afa3eb8/docs/client/)
+
+## Instance creation
+
+Creating an instance is done through CLI or through the (future) partner farm manager system. Some **settings** can be pre-defined on instance creation. ([doc](./instance.md#creation)).
+
+The CLI also allows to specify which source to use for `onboarding` and `home` applications. The defaults will be hosted on `github.com/cozy`.
+
+After creation, an instance has a `registerToken` generated randomly.
+
+All cozy-instance fields are considered private unless explictly prefixed by `public-`. Public instance field can be seen from the `instance` endpoint. Only public fields we can think of for now are `public-name`, `public-locale`.
+
+## Onboarding steps
+
+This document and the cozy-stack are only concerned with login and password registering step which are important for security.
+
+All other steps are handled by the `onboarding` application.
+
+The `onboarding` application SHOULD therefore provide the following features
+- When started with a `registerToken`, allow the user to create a password
+- When started with a `contextToken` ([see auth doc](./auth.md#how-to-get-a-token)) use it to retrieve instance document.
+  - If the instance document is complete **according to the `onboarding` app**, redirect to `home` application.
+  - Otherwise, performs whatever steps it deems necessary to fill out the instance (ask for user email, help set up `myaccounts` accounts, say thank you...)
+
+This makes cozy-stack simple and safer while allowing behaviour modification for several install types by picking the correct `onboarding` application / branch.
+
+This makes it easier to add more onboarding steps and have them run on already-installed cozy : On next login after onboarding application update, it will ask the user.
```

Comment:
Actually, it's a bit more difficult than that. For now, we decided to not run the new onboarding on already-installed cozy for functional reasons (as we don't have token usage, we reject new password saving in the unauthenticated way if the password is already set in the instance document). But I agree about adding more steps for "already-onboarded" cozies if we have token for safer process (and for `isRegistered` detection so).